staff/asier/react-spotify-api-promises/design/scripts/spotify-api-1.0.0.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Spotify API client.
+ *
+ * @version 1.0.0
+ */
+
+let spotifyApi;
+
+(function () {
+    "use strict";
+    const baseUrl = "https://api.spotify.com/v1/"
+    const token = 'BQCt_qBCV1kEnGnXS9KgyZpo5AJvcwfo3o1gPE6RJGz09CaZSJ9MUMd1pAjNnV8JxGH2kPMhZqQaNLdWGyGSYAyTgsQLBkBQjhtGe2kFph1PF_Rai6cPo-4k5nRnT4pbj-OJVAw'
+    const headers = { Authorization: 'Bearer ' + token }
+
+    
+    spotifyApi = {
+        baseUrl,

```

Comment:
WARN baseUrl is already defined and protected in a scope ref (baseUrl), which seems to have sense to avoid modifying it from outside. then why would you expose it here as a property if it can be accesible directly from the ref baseUrl wherever in any method? would it be convenient to avoid this property and just access to it from the "call" method directly, by making this one receiving a path instead of a full url. following code exposes this projection:

`       
function call(path) {    
           return fetch(`${baseUrl}/${path}`, { headers })
               .then(res => res.json())
       }
`
on the other hand, call should also be protected in a scoped function, but not a property of spotifyApi itself. and accesible from its methods directly by calling it: 

`
        searchArtists: query =>  call(`search?type=artist&q=${query}`).then(res =>  res.artists.items )
`

this avoids that any user from spotifyApi could ever use call method directly (avoid this: spotifyApi.call(...)...)

Reply:
not sure I fully understand with this warn and the one about above... Maybe we can review them tomorrow?